cycledash/auth.py

```diff
@@ -107,3 +108,33 @@ def register():
             login_user(user)
             return redirect('/')
     return render_template('register.html', errors=errors)
+
+def check_login_from_key(api_key):
+    username, password = api_key.split(":")
+    return check_login(username, password)
+
+def load_user_from_request(request):
+    """Support for basic authorization."""
+    # first, try to login using the api_key url arg
```

Comment:
Ah I just meant the url param version of this; keeping the BasicAuth version is :+1: 

Also, I'm confused a bit: I don't see an api_key urlparam in the download URL--how are you passing authorization though?